modules/component/src/let/let.directive.ts

```diff
@@ -117,26 +117,17 @@ export class LetDirective<U> implements OnDestroy {
   };
   private readonly updateViewContextObserver: Observer<U | null | undefined> = {
     next: (value: U | null | undefined) => {
-      // to have init lazy
-      if (!this.embeddedView) {
-        this.createEmbeddedView();
-      }
       this.viewContext.$implicit = value;
       this.viewContext.ngrxLet = value;
+      this.ensureEmbeddedViewCreated();
```

Comment:
Is there any reason why the `createEmbeddedView` is moved to the end of the callback?

Reply:
Yes, if the recursion happens you want to make sure you're setting the latest value last.

I'm trying to avoid the following problem.

`next(0)` called 
Call `this.createEmbeddedView()` 
Set `isEmbeddedviewCreated = true`
Call `viewContainerRef.createEmbeddedView()`
`next(1)` called
Call `this.createEmbeddedView()` 
return to `next(1)`
Set `this.viewContext.ngrxLet = value` (value = 1)
return to `viewContainerRef.createEmbeddedView`
return to `this.createEmbeddedView` 
return to `next(0)`
Set `this.viewContext.ngrxLet = value` (value = 0) OOPS!